// @hemant0621/animation/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.direction = exports.Animation = void 0;
const react_1 = require("react");
const scrollreveal_1 = __importDefault(require("scrollreveal"));
const useScrollReveal = (config = {}) => {
    const containerRef = (0, react_1.useRef)(null);
    (0, react_1.useEffect)(() => {
        if (containerRef.current) {
            (0, scrollreveal_1.default)().reveal(containerRef.current, config);
        }
        return () => {
            (0, scrollreveal_1.default)().destroy();
        };
    }, [config]);
    return containerRef;
};
function Animation(origin, distance, delay) {
    const containerRef = useScrollReveal({
        origin: origin,
        distance: distance,
        duration: 1000,
        delay: delay,
        easing: 'ease',
    });
    return containerRef;
}
exports.Animation = Animation;
function direction(origin) {
    const containerRef = useScrollReveal({
        origin: origin,
        distance: "30px",
        duration: 1000,
        delay: 200,
        easing: 'ease',
    });
    return containerRef;
}
exports.direction = direction;
